server/src/main/java/org/candlepin/resource/JobResource.java

```diff
@@ -186,46 +185,68 @@ public SchedulerStatus setSchedulerStatus(boolean running) {
         return getSchedulerStatus();
     }
 
-    @ApiOperation(notes = "Triggers select asynchronous jobs", value = "schedule")
+    @ApiOperation(notes = "Re-trigger cron jobs", value = "retrigger")
     @ApiResponses({ @ApiResponse(code = 400, message = ""), @ApiResponse(code = 500, message = "") })
     @POST
-    @Path("schedule/{task}")
+    @Path("retrigger/{task}")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @SuppressWarnings("unchecked")
-    public JobStatus schedule(@PathParam("task") String task) {
+    public void retrigger(@PathParam("task") String task) {
 
         /*
          * at the time of implementing this API, the only jobs that
          * are permissible are cron jobs.
          */
+        Class cronJobClass = getCronJobClass(task);
+
+        try {
+            pk.retriggerCronJob(task, cronJobClass);
+        }
+        catch (PinsetterException e) {
+            throw new IseException(i18n.tr("Error trying to schedule {0}: {1}", task,
+                e.getMessage()));
+        }
+    }
+
+    @ApiOperation(notes = "Fires cron jobs asynchronously and immediately", value = "schedule")
+    @ApiResponses({ @ApiResponse(code = 400, message = ""), @ApiResponse(code = 500, message = "") })
+    @POST
+    @Path("schedule/{task}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @SuppressWarnings("unchecked")
+    public JobStatus schedule(@PathParam("task") String task) {
+
+        Class cronJobClass = getCronJobClass(task);
+        try {
+            /*
+             * at the time of implementing this API, the only jobs that
+             * are permissible are cron jobs.
+             */
+            return pk.scheduleSingleJob(cronJobClass, Util.generateUUID());
+        }
+        catch (PinsetterException e) {
+            throw new IseException(i18n.tr("Error trying to schedule {0}: {1}", task,
+                e.getMessage()));
+        }
+    }
+
+    private Class getCronJobClass(String task) {
         String className = "org.candlepin.pinsetter.tasks." + task;
         try {
-            boolean allowed = false;
             for (String permissibleJob : ConfigProperties.DEFAULT_TASK_LIST) {
                 if (className.equalsIgnoreCase(permissibleJob)) {
-                    allowed = true;
-                    // helps ignore case
-                    className = permissibleJob;
+                    return Class.forName(permissibleJob);
```

Comment:
If this fails, it will throw a ClassNotFoundException, right?  I don't really see how the code would ever even make it to the `throws` line you have at the end of the method.

Reply:
@awood lets talk about this online.

Reply:
talked with @awood , no changes necessary.